x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_form.tsx

```diff
@@ -240,49 +243,49 @@ export const ActionForm = ({
+      actionTypesIndex && actionTypesIndex[actionTypeModel.id]?.isSystemActionType
+    );
     let actionTypeConnectors = connectors.filter(
       (field) => field.actionTypeId === actionTypeModel.id
     );
 
-    if (actionTypeConnectors.length > 0) {
-      actions.push({
-        id: '',
-        actionTypeId: actionTypeModel.id,
-        group: defaultActionGroupId,
-        params: {},
-        frequency: defaultRuleFrequency,
-        uuid: uuidv4(),
-      });
-      setActionIdByIndex(actionTypeConnectors[0].id, actions.length - 1);
-    } else {
-      actionTypeConnectors = connectors.filter((field) =>
-        allowGroupConnector.includes(field.actionTypeId)
-      );
-      if (actionTypeConnectors.length > 0) {
-        actions.push({
+    const actionToPush = isSystemAction
+      ? {
           id: '',
-          actionTypeId: actionTypeConnectors[0].actionTypeId,
+          actionTypeId: actionTypeModel.id,
+          params: {},
+          uuid: uuidv4(),
+          type: RuleActionTypes.SYSTEM,
+        }
+      : {
+          id: '',
+          actionTypeId: actionTypeModel.id,
           group: defaultActionGroupId,
           params: {},
-          frequency: DEFAULT_FREQUENCY,
+          frequency: defaultRuleFrequency,
           uuid: uuidv4(),
-        });
-        setActionIdByIndex(actionTypeConnectors[0].id, actions.length - 1);
-      }
-    }
+          type: RuleActionTypes.DEFAULT,
+        };
 
     if (actionTypeConnectors.length === 0) {
-      // if no connectors exists or all connectors is already assigned an action under current alert
-      // set actionType as id to be able to create new connector within the alert form
-      actions.push({
-        id: '',
-        actionTypeId: actionTypeModel.id,
-        group: defaultActionGroupId,
-        params: {},
-        frequency: defaultRuleFrequency,
-      });
-      setActionIdByIndex(actions.length.toString(), actions.length - 1);
-      setEmptyActionsIds([...emptyActionsIds, actions.length.toString()]);
+      actionTypeConnectors = connectors.filter((field) =>
+        allowGroupConnector.includes(field.actionTypeId)
+      );
+      if (actionTypeConnectors.length > 0) {
+        // If a connector was successfully found, update the actionTypeId
+        actions.push({ ...actionToPush, actionTypeId: actionTypeConnectors[0].actionTypeId });
+        setActionIdByIndex(actionTypeConnectors[0].id, actions.length - 1);
+      } else {
+        // if no connectors exists or all connectors is already assigned an action under current alert
+        // set actionType as id to be able to create new connector within the alert form
+        actions.push(actionToPush);
+        setActionIdByIndex(actions.length.toString(), actions.length - 1);
+        setEmptyActionsIds([...emptyActionsIds, actions.length.toString()]);
+      }
+    } else {
+      actions.push(actionToPush);
+      setActionIdByIndex(actionTypeConnectors[0].id, actions.length - 1);
```

Comment:
Let's add some jest tests for this

Reply:
This is not new functionality. This is a refactor of an existing control flow at @XavierM's request (https://github.com/elastic/kibana/pull/175057#discussion_r1462506168) and if it wasn't already covered by tests, I can't add that to the scope of this PR. Please open a new issue.

Reply:
Tracking this here: https://github.com/elastic/kibana/issues/175839